security/security.c

```diff
@@ -464,6 +469,45 @@ static int lsm_append(const char *new, char **result)
 	return 0;
 }
 
+#define LSM_FUNC_DEFAULT(NAME)		NAME##_func_default
+#define HOOK_STATIC_CALL(HOOK, NUM)	static_call_##HOOK##_##NUM
+#define HOOK_STATIC_CHECK(HOOK, NUM)	static_check_##HOOK##_##NUM
+
+#define FOR_EACH_HOOK_SLOT(M, ...)			\
+	M(1, __VA_ARGS__)				\
+	M(2, __VA_ARGS__)				\
+	M(3, __VA_ARGS__)
+
+#define CREATE_STATIC(NUM, NAME)			\
+	DEFINE_STATIC_CALL(HOOK_STATIC_CALL(NAME, NUM), LSM_FUNC_DEFAULT(NAME));	\
+	DEFINE_STATIC_KEY_FALSE(HOOK_STATIC_CHECK(NAME, NUM));
+
+// We need a default function that will not be called so that 
+// static_call can infer the expected type
+#define LSM_HOOK(RET, DEFAULT, NAME, ...) 		\
+	noinline RET LSM_FUNC_DEFAULT(NAME)(__VA_ARGS__)\
+	{						\
+		return DEFAULT;				\
+	}						\
```

Comment:
This function is defined but never used except for getting its type. We might be able to get rid of them by modifying the static calls.